source/1_engine/rex_engine/include/rex_engine/filesystem/vfs.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "rex_engine/filesystem/mount_roots.h"
+#include "rex_engine/types.h"
+#include "rex_std/bonus/functional.h"
+#include "rex_std/functional.h"
+#include "rex_std/memory.h"
+#include "rex_std/string_view.h"
+#include "rex_std/type_traits.h"
+#include "rex_std_extra/utility/type_id.h"
+#include "rex_std_extra/utility/yes_no.h"
+
+namespace rex::vfs
+{
+  DEFINE_YES_NO_ENUM(AppendToFile);
+
+  class QueuedRequest;
+
+  class ReadRequest
+  {
+  public:
+    ReadRequest(rsl::string_view filepath, QueuedRequest* queuedRequest);
+    ReadRequest(const ReadRequest& other);
+    ReadRequest(ReadRequest&& other);
+
+    ~ReadRequest();
+
+    ReadRequest& operator=(const ReadRequest& other);
+    ReadRequest& operator=(ReadRequest&& other);
+
+    void signal(const char8* buffer, count_t count);
+    void wait() const;
+
+    const char8* data() const;
+    count_t count() const;
+
+    rsl::string_view filepath() const;
+
+  private:
+    rsl::string_view m_filepath;
```

Comment:
Do we need to store the file path?
Is there a use for retrieving the file path?

Reply:
it's for debugging purposes